File: in_C++/1814A_coins.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    while(t--) {
        long long num,k;
        cin >> num >> k;

        if (num % 2 == 1 and k % 2 == 0) {
            cout << "NO" << endl;
        }
        else if (num == k) {
            cout << "Yes" << endl;
        }
        else if (num % 2 ==0) {
            cout << "YES" << endl;
        }
        else if ( num % 2 == 1 and k % 2 == 1) {
            cout << "YEs" << endl;
        }
    }
    return 0;
}

File: in_Python/1814A_coins.py
t = int(input())
for i in range(t):
    num,k = map(int,input().split())
    if num % 2 == 1 and k % 2 == 0:
        print("NO")
    elif num % 2 == 0:
        print("YES")
    elif num == k:
        print("YES")
    elif num % 2 == 1 and k % 2 == 1:
        print("YES")